drivers: watchdog: Enable Gecko wdt driver efr32bg_sltb010a

fkokosinski opened the pull request:
This commit enables the Gecko Watchdog Timer driver on the efr32bg_sltb010 board.

martinjaeger commented on drivers/watchdog/wdt_gecko.c:
```diff
@@ -246,11 +246,6 @@ static int wdt_gecko_init(const struct device *dev)
 {
 	const struct wdt_gecko_cfg *config = dev->config;
 
-#ifdef CONFIG_WDT_DISABLE_AT_BOOT
-	/* Ignore any errors */
-	wdt_gecko_disable(dev);
-#endif
-
```
What's the reason for moving this further down?

kgugala replied:
this seems to be some kind of leftover after splitting the work into smaller chunks for PRs. Fixed

martinjaeger commented on drivers/watchdog/wdt_gecko.c:
```diff
@@ -259,6 +254,13 @@ static int wdt_gecko_init(const struct device *dev)
+	CMU_ClockEnable(cmuClock_WDOG0, true);
+#endif
+
+#if IS_ENABLED(CONFIG_WDT_DISABLE_AT_BOOT)
```
The `IS_ENABLED` macro is meant for `if` statements in the code and not macros, so let's keep the more simple approach as before:

```suggestion
#ifdef CONFIG_WDT_DISABLE_AT_BOOT
```

kgugala replied:
fixed

martinjaeger commented on drivers/watchdog/wdt_gecko.c:
```diff
@@ -259,6 +254,13 @@ static int wdt_gecko_init(const struct device *dev)
 #else
 	CMU_ClockSelectSet(config->clock, cmuSelect_ULFRCO);
+	/* Enable Watchdog clock. */
+	CMU_ClockEnable(cmuClock_WDOG0, true);
```
Why not use the `CMU_ClockEnable(config->clock, true);` a few lines above?

kgugala replied:
yup, updated the code

msierszulski committed:
drivers: watchdog: Enable Gecko wdt driver efr32bg_sltb010a

This commit enables the Gecko Watchdog Timer driver on the
efr32bg_sltb010 board.

Signed-off-by: Mateusz Sierszulski <msierszulski@antmicro.com>

martinjaeger commented on drivers/watchdog/wdt_gecko.c:
```diff
@@ -259,6 +259,8 @@ static int wdt_gecko_init(const struct device *dev)
 #else
 	CMU_ClockSelectSet(config->clock, cmuSelect_ULFRCO);
+	/* Enable Watchdog clock. */
+	CMU_ClockEnable(config->clock, true);
```
One more simplification. As the clock is now enabled in both cases, we can take it out of the `#ifdef`s.

```diff
- #if !defined(_SILICON_LABS_32B_SERIES_2)
- 	/* Ensure LE modules are clocked */
- 	CMU_ClockEnable(config->clock, true);
- #else
+ #ifdef _SILICON_LABS_32B_SERIES_2
	CMU_ClockSelectSet(config->clock, cmuSelect_ULFRCO);
+ #endif
+ 
	/* Enable Watchdog clock. */
	CMU_ClockEnable(config->clock, true);
- #endif
```

I don't know this MCU at all. Is it correct that the `CMU_ClockSelectSet` call is only necessary for a particular series? The `ULFRCO` is enabled for all cases, so this looks a bit strange. Can you please double-check?

lpawelcz replied:
Simplified the code as requested

martinjaeger reviewed (dismissed):
Looks good from watchdog driver perspective, but I don't have any experience with this type of MCU, so I can't judge the implementation details.

kgugala reviewed (approved):
LGTM
